javascript-practice/src/js/views/dashboardView.js

```diff
@@ -78,18 +115,115 @@ export default class DashboardView {
       const valueAddPassword = this.addPasswordEl.value;
       const valueAddConfirmPassword = this.addConfirmPasswordEl.value;
 
-      submitAddUser(
-        valueAddUserId,
-        valueAddFirstNameId,
-        valueAddLastNameId,
-        valueAddEmailId,
-        valueAddMobileNoId,
-        valueAddRole,
-        valueAddUserName,
-        valueAddPassword,
-        valueAddConfirmPassword
-      );
+      Object.values(this.addUserErrorEls).forEach((el) => {
+        el.textContent = '';
+      });
+
+      if (
+        !valueAddUserId ||
+        !valueAddFirstNameId ||
+        !valueAddLastNameId ||
+        !valueAddEmailId ||
+        !valueAddMobileNoId ||
+        !valueAddUserName ||
+        !valueAddPassword ||
+        !valueAddConfirmPassword
+      ) {
+        Object.entries(this.addUserErrorEls).forEach(([key, el]) => {
+          if (!this[key].value) {
+            el.textContent = `${REQUIRED_FIELD}`;
+          }
+        });
+
+        return;
+      }
+      let isValid = true;
+      if (isValid) {
+        if (
+          !validateInputLength(valueAddUserId) ||
+          !validateInputLength(valueAddFirstNameId) ||
+          !validateInputLength(valueAddLastNameId) ||
+          !validateInputLength(valueAddUserName)
+        ) {
+          if (
+            !validateInputLength(valueAddUserId) &&
+            this.addUserErrorEls.addUserIdEl
+          )
+            this.addUserErrorEls.addUserIdEl.textContent = `${REQUIRED_FIELD_THREE}`;
+          if (
+            !validateInputLength(valueAddFirstNameId) &&
+            this.addUserErrorEls.addFirstNameEl
+          )
+            this.addUserErrorEls.addFirstNameEl.textContent = `${REQUIRED_FIELD_THREE}`;
+          if (
+            !validateInputLength(valueAddLastNameId) &&
+            this.addUserErrorEls.addLastNameEl
+          )
+            this.addUserErrorEls.addLastNameEl.textContent = `${REQUIRED_FIELD_THREE}`;
+          if (
+            !validateInputLength(valueAddUserName) &&
+            this.addUserErrorEls.addUserNameEl
+          )
+            this.addUserErrorEls.addUserNameEl.textContent = `${REQUIRED_FIELD_THREE}`;
+          isValid = false;
+        }
+      }
+
+      if (!valueAddRole) {
+        this.addUserErrorEls.addRoleEl.textContent = `${REQUIRED_ROLE}`;
+        isValid = false;
+      }
+
+      if (!validatePhoneNumber(valueAddMobileNoId)) {
+        this.addUserErrorEls.addMobileNoEl.textContent = `${INVALID_PHONE_NUMBER}`;
+
+        return;
+      }
+
+      if (!validatePassword(valueAddPassword)) {
+        this.addUserErrorEls.addPasswordEl.textContent = `${INVALID_PASSWORD}`;
+
+        return;
```

Comment:
Code DO NOT the same style. Somewhere you return, somewhere you set isValid. Could you explain?

Reply:
i done fix